File: src/console/log/logger/PrefixedLogger.php
<?php

namespace pocketcloud\cloud\console\log\logger;

use pocketcloud\cloud\console\log\level\CloudLogLevel;

final class PrefixedLogger extends MainLogger {

    public function __construct(
        private readonly MainLogger $logger,
        private string $prefix
    ) {
        parent::__construct(null, false, false);
        $this->close();
    }

    public function log(CloudLogLevel $logLevel, string $message, ...$params): MainLogger {
        return $this->logger->log($logLevel, $this->prefix . " $message", ...$params);
    }

    public function setPrefix(string $prefix): void {
        $this->prefix = $prefix;
    }

    public function getPrefix(): string {
        return $this->prefix;
    }
}

File: src/main/java/de/pocketcloud/cloud/console/log/def/PrefixedLogger.java
package de.pocketcloud.cloud.console.log.def;

import de.pocketcloud.cloud.console.log.CloudLogLevel;
import lombok.Getter;
import lombok.Setter;

public class PrefixedLogger extends MainLogger {

    private final MainLogger parentLogger;
    @Setter
    @Getter
    private String prefix;

    public PrefixedLogger(MainLogger parentLogger, String prefix) {
        super(null, false, false);
        this.parentLogger = parentLogger;
        this.prefix = prefix;
        this.closeLogFile();
    }

    @Override
    public PrefixedLogger log(CloudLogLevel logLevel, String message, Object... params) {
        parentLogger.log(logLevel, prefix + " " + message, params);
        return this;
    }
}